docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,707 @@
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+
+If the schema of the data is not known at compile time, or too cumbersome, you can revert to the dynamically-typed API: 
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Create the reader: define the data columns and where to find them in the text file.
+var reader = new TextLoader(env, new TextLoader.Arguments
+{
+    Column = new[] {
+	    // We read the first 10 values as a single float vector.
+        new TextLoader.Column("FeatureVector", DataKind.R4, new[] {new TextLoader.Range(0, 9)}),
+        // Separately, read the target variable.
+        new TextLoader.Column("Target", DataKind.R4, 10)
+    },
+    // Default separator is tab, but we need a comma.
+    Separator = ","
+});
+
+// Now read the file (remember though, readers are lazy, so the actual reading will happen when the data is accessed).
+var data = reader.Read(new MultiFileSource(dataPath));
+```
+
+## How do I look at the intermediate data?
+
+Oftentimes, when we construct the experiment, we want to make sure that the data processing 'up to a certain moment' produces the results that we want. With ML.NET it is not very easy to do: since all ML.NET operations are lazy, the objects we construct are just 'promises' of data.
+
+We will need to create the cursor and scan the data to obtain the actual values. One way to do this is to use [schema comprehension](SchemaComprehension.md) and map the data to an `IEnumerable` of user-defined objects.
+
+Another mechanism that lets you inspect the intermediate data is the `GetColumn<T>` extension method. It lets you look at the contents of one column of your data in a form of an `IEnumerable`.
+
+Here is all of this in action:
+
+Example file (https://github.com/dotnet/machinelearning/blob/master/test/data/adult.tiny.with-schema.txt):
+```
+Label	Workclass	education	marital-status
+0	Private	11th	Never-married
+0	Private	HS-grad	Married-civ-spouse
+1	Local-gov	Assoc-acdm	Married-civ-spouse
+1	Private	Some-college	Married-civ-spouse
+
+```
+
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Create the reader: define the data columns and where to find them in the text file.
+var reader = TextLoader.CreateReader(env, ctx => (
+        // A boolean column depicting the 'target label'.
+        IsOver50K: ctx.LoadBool(0),
+        // Three text columns.
+        Workclass: ctx.LoadText(1),
+        Education: ctx.LoadText(2),
+        MaritalStatus: ctx.LoadText(3)),
+    hasHeader: true);
+
+// Start creating our processing pipeline. For now, let's just concatenate all the text columns
+// together into one.
+var dataPipeline = reader.MakeNewEstimator()
+    .Append(row =>
+        (
```

Comment:
>         [](start = 0, length = 8)

Not a big deal, but this seems to be inconsistent with your prior syntax where you seem to have preferred `=> ( #newline# content` vs. `=> #newline# ( #newline# content` as you do here.